backend/matchmaker_service.py

```diff
@@ -0,0 +1,170 @@
+import firebase_admin
+from firebase_admin import credentials, firestore
+from game import Game
+from game_engine.common import GameOptions, GameSchedule, STV_I18N_TABLE, ISODayOfWeek, log_message
+from game_engine.database import Database
+from game_engine.engine import Engine
+from game_engine.firestore import FirestoreDB
+from game_engine.matchmaker import MatchMakerInterface
+from test_game import MockDatabase, MockPlayEngine
+from google.cloud.firestore_v1.document import DocumentSnapshot
+import time
+import threading
+import datetime
+from game_engine import events
+from multiprocessing import Process
+import multiprocessing
+
+_FIRESTORE_PROD_CONF_JSON_PATH = ''
+_TEST_FIRESTORE_INSTANCE_JSON_PATH = '../firebase/stv-game-db-test-4c0ec2310b2e.json'
+json_config_path = _TEST_FIRESTORE_INSTANCE_JSON_PATH
+_AMAZON_SQS_PROD_CONF_JSON_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_AMAZON_SQS_CONFIG_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_TWILIO_SMS_CONFIG_PATH = '../twilio/stv-twilio.json'
+
+
+
+class MatchmakerService:
+    # Handles scheduling and communication with other services for starting games
+    # TDOO(David): Add function to run all games that are supposed to be running at start(in MVP/test)
+    def __init__(self, matchmaker: MatchMakerInterface, gamedb: Database, json_config_path=json_config_path, region="US", min_players=5, is_mvp=True):
+        self._matchmaker = matchmaker
+        self._gamedb = gamedb
+        self._min_players = min_players
+        self._region = region
+        self._is_mvp = is_mvp
+        self._stop = threading.Event()
+        self._daemon_started = False
+
+    def play_game(self, game: Game, players:list, game_dict:dict, is_test=True):
+        log_message("Starting a game", game_id=game_dict.get("id"), additional_tags=game_dict)
+
+        if is_test:
+            database = MockDatabase()
+            engine = MockPlayEngine().CreateEngine(database)
+        else:
+            database = FirestoreDB(json_config_path=json_config_path, game_id=game._game_id)#db needs to have correct game_id
+            engine = Engine(options=game._options,
+                            game_id=game._game_id,
+                            sqs_config_path=_TEST_AMAZON_SQS_CONFIG_PATH,
+                            twilio_config_path=_TEST_TWILIO_SMS_CONFIG_PATH,
+                            gamedb=database
+            )
+
+        game_data = self._matchmaker.generate_teams_tribes(game_id=game._game_id, players=players, game_options=game._options, gamedb=database)
+        tribes = game_data['tribes']
+        # TODO notify users that game started 

```

Comment:
let's add a method for the notify logic and call it here. in the method body we can leave the TODO.